### DISCUSSION TITLE
Feature Request: "Eager" autolink

### DISCUSSION THREAD

**Discussion by dcourv** [ueberdosis/tiptap#4711] [Feature Requests]
### Description

It would be great if there were a setting where where autolinks were set or unset immediately after the text becomes a link or is no longer a link, e.g.

google.c| <- cursor, text is not a link
type 'o',
google.co| <- cursor, text becomes a link
type <backspace>
google.c| <- cursor, text is not a link

Currently, you have to insert whitespace after the link for it to be autolinked, and autolinks are not unset when their text is no longer a link.

### Use Case

If users are typing short rich-styled posts or comments where styling controls are minimal, this might let them know that links will be autolinked more easily.

This behavior is fairly common and something that users would likely want or expect. For example, it is default behavior in the Twitter/X post editor, as well as in Remirror.

I could see users wanting either the current behavior (which is the default in Notion) or the "eager" autolink behavior described above, so it would be great to have a setting for this!

### Type

New feature

**Comment by bdbch** [comment]
I've seen your PR. It works if you don't consider adding new characters to an existing link. I agree that empty text links should be cleaned up but the `inclusive: false` removes the option to add any character at the end of the link mark.

**Comment by dcourv** [comment]
Hi @bdbch. This feature request is related but different from the PR, it's actually about when the text becomes a link when autolink is enabled (e.g. after typing 'o' in google.com, or after typing the space after google.com) and when the link is automatically unset. Please see the above for a more detailed description.
